crates/events/tests/web.rs

```diff
@@ -50,29 +49,29 @@ impl<A> Sender<A> {
     }
 }
 
-fn mpsc<A, F>(f: F) -> impl Future<Item = Vec<A>, Error = JsValue>
+async fn mpsc<A, F>(f: F) -> Vec<A>
 where
     F: FnOnce(Sender<A>),
 {
-    let (sender, receiver) = futures::sync::mpsc::unbounded();
+    let (sender, receiver) = futures::channel::mpsc::unbounded();
 
     f(Sender { sender });
 
     receiver
-        .then(|x| match x {
+        .map(|x| match x {
```

Comment:
This doesn't seem right, it's not handling the `Result` properly. It should be something like this instead:

```rust
receiver.try_collect().await
```